### PR TITLE
Add Choreo Connect Policy Specs and Definitions

### PR DESCRIPTION
**Pull request by renuka-fernando**
$suject
fix https://github.com/wso2/product-apim/issues/12507

### Related PRs
Update file extention: https://github.com/wso2/carbon-apimgt/pull/11089
OPA spec: https://github.com/wso2/product-apim/pull/12499

### CODE REVIEW

File: modules/distribution/resources/operation_policies/specifications/ccCallInterceptorService.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "name": "ccCallInterceptorService",
+  "displayName": "Call Interceptor Service",
+  "description": "Call an interceptor service for message mediation",
+  "multipleAllowed": true,
+  "policyAttributes": [
+    {
+      "name": "interceptorServiceURL",
+      "displayName": "Service URL",
+      "description": "Interceptor Service URL",
+      "validationRegex": "^(([\\w+]+\\:\\/\\/)?([\\w\\d-]+\\.)*[\\w-]+([\\.\\:]\\w+)*([\\/\\?\\=\\&\\#\\.]?[\\w-]+)*\\/?)$",
```
Comment (renuka-fernando) [inline modules/distribution/resources/operation_policies/specifications/ccCallInterceptorService.json:12]: todo: check regex with the OPA server URL

File: modules/distribution/resources/operation_policies/definitions/opaPolicy.gotmpl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+definition:
+  action: OPA
+  parameters:
+    requestGenerator: ""
```
Comment (AmaliMatharaarachchi) [inline modules/distribution/resources/operation_policies/definitions/opaPolicy.gotmpl:4]: why don't we template this?

File: modules/distribution/resources/operation_policies/specifications/ccAddHeader.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "applicableFlows": [
+    "request"
+  ],
+  "supportedGateways": [
+    "ChoreoConnect"
+  ],
+  "supportedApiTypes": [
+    "HTTP"
+  ]
+}
```
Comment (AmaliMatharaarachchi) [inline modules/distribution/resources/operation_policies/specifications/ccAddHeader.json:34]: let's add new line at EOL

File: modules/distribution/resources/operation_policies/definitions/opaPolicy.gotmpl (hunk 2 above)
Comment (renuka-fernando) [reply modules/distribution/resources/operation_policies/definitions/opaPolicy.gotmpl:4]: We do not get this from UI, so this is not available in the policy spec.
If someone wants to have a custom requestGenerator generator, they have to create a new policy, by adding the class name here.

File: modules/distribution/resources/operation_policies/specifications/ccAddHeader.json (hunk 3 above)
Comment (renuka-fernando) [reply modules/distribution/resources/operation_policies/specifications/ccAddHeader.json:34]: these JSONs do not contain EOF

**Review by 1akshitha — APPROVED** [review]

File: modules/distribution/resources/operation_policies/definitions/ccChangeHTTPMethod.gotmpl [hunk 4]
```diff
@@ -0,0 +1,5 @@
+definition:
+  action: REWRITE_RESOURCE_METHOD
+  parameters:
+    currentMethod: {{ .currentMethod }}
+    updatedMethod: {{ .updatedMethod }}
```
[the remark this answers is not in the document]
Comment (pubudu538) [reply modules/distribution/resources/operation_policies/definitions/ccChangeHTTPMethod.gotmpl:5]: This is fine.

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**